ddclient

```diff
@@ -4670,8 +4685,82 @@ sub nic_woima_update {
             if $state ne 'results2';
     }
 }
+######################################################################
+## nic_dnsexit_examples
+######################################################################
+sub nic_dnsexit_examples {
+    return <<"EoEXAMPLE";
+o 'dnsexit'
+
+The 'dnsexit' protocol is the protocol used by the dynamic hostname services
+of the 'DnsExit' dns services. This is currently used by the free
+dynamic DNS service offered by www.dnsexit.com.
 
+Configuration variables applicable to the 'dnsexit' protocol are:
+  ssl=no                       ## turn off ssl
+  protocol=dnsexit             ##
+  server=update.dnsexit.com    ## defaults to update.dnsexit.com
+  use=web                      ## defaults to web
+  web=update.dnsexit.com       ## defaults to update.dnsexit.com
+  script=/RemoteUpdate.sv      ## defaults to /RemoteUpdate.sv
+  login=service-userid         ## userid registered with the service
+  password=service-password    ## password registered with the service
+  fully.qualified.host         ## the host registered with the service.
 
+Example ${program}.conf file entries:
+  ## single host update
+  protocol=dnsexit             \\ 
+  login=service-userid         \\
+  password=service-password    \\
+  fully.qualified.host
+
+EoEXAMPLE
+}
+######################################################################
+## nic_dnsexit_update
+######################################################################
+sub nic_dnsexit_update {
+    debug("\nnic_dnsexit_update -------------------");
+
+    ## update each configured host
+    foreach my $h (@_) {
+    my $ip = delete $config{$h}{'wantip'};
+        info("setting IP address to %s for %s", $ip, $h);
+        verbose("UPDATE:","updating %s", $h);
+		
+        # Set the URL that we're going to update
+        my $url;
+        $url  = "http://$config{$h}{'server'}$config{$h}{'script'}";
+        $url .= "?login=$config{$h}{'login'}";
+        $url .= "&password=$config{$h}{'password'}";
+        $url .= "&host=$h";
+        $url .= "&myip=$ip";
+
+        # Try to get URL
+        my $reply = geturl(opt('proxy'), $url);
+
+        # No response, declare as failed
+        if (!defined($reply) || !$reply) {
+            failed("updating %s: Could not connect to %s.", $h, $config{$h}{'server'});
+            last;
+        }
+        last if !header_ok($h, $reply);
+
+        # Response found
+        if ($reply =~ /(\d+)=(.+)/)
```

Comment:
According to the documentation at https://www.dnsexit.com/Direct.sv?cmd=ipClients, This regex will match whether there is a success or error.

Reply:
take a look at the patch i (truesalo) sent. it fixes this.

Reply:
I seen that, and I agree fixes the problem.  Doesn't seem it was merged into this pull request though.